File: cpp/1-multiples-3-5.cpp
#include <iostream>

using std::cout;
using std::endl;

int sum_of_multiples(int multiple, int max_no);

int main() {
    int number1 = 3;
    int number2 = 5;
    int product = 15;
    int max_no = 999;

    int sum = sum_of_multiples(number1, max_no) + sum_of_multiples(number2, max_no) - sum_of_multiples(product, max_no);

    cout<<"Sum: "<<sum;

    return 0;
}

int sum_of_multiples(int multiple, int max_no) {
    int quotient = max_no / multiple;
    int sum = quotient * (quotient + 1) / 2;

    return sum * multiple;
}

File: rust/1-multiples-3-5.rs
fn main() {
    let max_no: i32 = 999;
    let first_number: i32 = 3;
    let second_number: i32 = 5;
    let product: i32 = first_number * second_number;

    let sum: i32 = sum_of_products(first_number, max_no) + sum_of_products(second_number, max_no) - sum_of_products(product, max_no);

    println!("Sum: {}", sum);
}

fn sum_of_products(multiple: i32, max_no: i32) -> i32 {
    let quotient: i32 = max_no / multiple;
    let sum: i32 = quotient * (quotient + 1) / 2;

    return multiple * sum;
}
